src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreGraph.cs

```diff
@@ -135,8 +135,7 @@ public static RestoreTargetGraph Create(bool writeToLockFile, IEnumerable<GraphN
 
                         if (string.Equals(node?.Item?.Key?.Type, LibraryTypes.Unresolved))
                         {
-                            if (node.Key.TypeConstraint != LibraryDependencyTarget.Reference
-                                && node.Key.VersionRange != null)
+                            if (node.Key.VersionRange != null)
```

Comment:
Does this work if a version is defined by the framework reference? or is that no longer flowing through here?